File: April 2020/09 - Backspace String Compare/Solution.py
class Solution:
    def backspaceCompare(self, S: str, T: str) -> bool:
        def Build(S):
            ans = []
            for c in S:
                if c != "#":
                    ans.append(c)
                elif ans:
                    ans.pop()
            return "".join(ans)
        return Build(S) == Build(T)

File: April 2020/09 - Backspace String Compare/Solution.cpp
class Solution {
public:
     string Build(string x) {
        string str = "";
        for(int i = 0; i < x.size(); ++i) {
            if(x[i] != '#')
                str += x[i];
            else if(!str.empty())
                str.pop_back();
        }
        return str;
    }
    
    bool backspaceCompare(string S, string T) {
        return Build(S) == Build(T);
    }
};